lib/core.ps1

```diff
@@ -604,21 +614,63 @@ set args=%args:(=``(%
+) else (
+    powershell -noprofile -ex unrestricted -command `"& '$resolved_path' $arg %args%;exit `$lastexitcode`"
+)" | Out-File "$shim.cmd" -Encoding ASCII
+
+        warn_on_overwrite $shim $path
+        "#!/bin/sh
+# $resolved_path
+if command -v pwsh.exe &> /dev/null; then
+    pwsh.exe -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"
+else
+    powershell.exe -noprofile -ex unrestricted -command `"& '$resolved_path' $arg $@;exit \`$lastexitcode`"
+fi" | Out-File $shim -Encoding ASCII
+    } elseif ($path -match '\.jar$') {
+        warn_on_overwrite "$shim.cmd" $path
+        "@rem $resolved_path
+@java -jar `"$resolved_path`" $arg %*" | Out-File "$shim.cmd" -Encoding ASCII
+
+        warn_on_overwrite $shim $path
+        "#!/bin/sh
+# $resolved_path
+java -jar `"$resolved_path`" $arg `"$@`"" | Out-File $shim -Encoding ASCII
+    } elseif ($path -match '\.py$') {
+        warn_on_overwrite "$shim.cmd" $path
+        "@rem $resolved_path
+@python `"$resolved_path`" $arg %*" | Out-File "$shim.cmd" -Encoding ASCII
+
+        warn_on_overwrite $shim $path
+        "#!/bin/sh
+# $resolved_path
+python `"$resolved_path`" $arg `"$@`"" | Out-File $shim -Encoding ASCII
+    } else {
+        warn_on_overwrite "$shim.cmd" $path
+        # find path to Git's bash so that batch scripts can run bash scripts
+        $gitdir = (Get-Item (Get-Command git -ErrorAction:Stop).Source -ErrorAction:Stop).Directory.Parent
+        if ($gitdir.FullName -imatch 'mingw') {
+            $gitdir = $gitdir.Parent
+        }
+        "@rem $resolved_path
+@`"$(Join-Path (Join-Path $gitdir.FullName 'bin') 'bash.exe')`" `"$resolved_path`" $arg %*" | Out-File "$shim.cmd" -Encoding ASCII
 
-        "#!/bin/sh`npowershell.exe -noprofile -ex unrestricted `"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
-    } elseif($path -match '\.jar$') {
-        "@java -jar `"$resolved_path`" $arg %*" | out-file "$shim.cmd" -encoding ascii
-        "#!/bin/sh`njava -jar `"$resolved_path`" $arg `"$@`"" | out-file $shim -encoding ascii
+        warn_on_overwrite $shim $path
+        "#!/bin/sh
+# $resolved_path
+`"$resolved_path`" $arg `"$@`"" | Out-File $shim -Encoding ASCII
     }
 }
 
 function get_shim_path() {
-    $shim_path = "$(versiondir 'scoop' 'current')\supporting\shimexe\bin\shim.exe"
+    $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\kiennq\shim.exe"
     $shim_version = get_config 'shim' 'default'
     switch ($shim_version) {
         '71' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\71\shim.exe"; Break }
-        'kiennq' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shims\kiennq\shim.exe"; Break }
+        'scoopcs' { $shim_path = "$(versiondir 'scoop' 'current')\supporting\shimexe\bin\shim.exe"; Break }
+        'kiennq' { Break } # for backward compatibility
         'default' { Break }
```

Comment:
It seems there is no way for the user to select the `shimexe\bin\shim.exe` shim?

Reply:
`scoop config shim scoopcs` does that.

Reply:
Oh, duh. It's right there, sorry.